ldp/nn/handlers/chunking.py

```diff
@@ -0,0 +1,170 @@
+from typing import TypeVar
+
+import torch
+from transformers.generation.utils import GenerateDecoderOnlyOutput
+
+TOutputType = TypeVar("TOutputType", torch.Tensor, GenerateDecoderOnlyOutput)
+
+
+class TensorChunker:
+    """Splits tensors into chunks and adds dummy chunks as needed for parallel processing frameworks like FSDP."""
+
+    def __init__(self, num_chunks: int, dummy_value: int = 0):
+        self.num_chunks = num_chunks
+        self.dummy_value = dummy_value
+
+    def chunkify(self, *args, **kwargs) -> tuple[list[tuple], list[dict], list[bool]]:
+        """Splits the args into self.num_chunks chunks, adding dummy chunks as needed.
+
+        Returns:
+            tuple: Contains:
+                - A list of len self.num_chunks, where each element is a tuple of split args.
+                - A list of len self.num_chunks, where each element is a dict of split kwargs.
+                - A list of len self.num_chunks, where each element is a bool indicating
+                  whether the chunk is a dummy chunk.
+
+        NOTE: the below Note is not always true - should come back and fix.
+            Can be seen by observing e.g. `len(torch.chunk(torch.arange(20), 8, dim=0)) == 7`.
+            Not necessarily a huge problem - just means we'll be chunking suboptimally.
+            But there may be cases where that causes problems.
+        # Note: Dummy flags will be set only if the incoming batch is smaller than self.num_chunks,
+        # resulting in empty chunks insertion.
+
+        Example:
+        ```python
+        input_tensor = torch.randn(10, 5)  # A tensor with batch size 10
+        some_int = 42  # An integer value that isn't split
+
+        chunker = TensorChunker(num_chunks=4)
+        split_args, split_kwargs, dummy_flags = chunker.chunkify(
+            input_tensor=input_tensor, some_int=some_int
+        )
+        ```
+
+        Output split_args:
+        [
+            ((input_tensor_chunk_0, ), (some_int, )),  # batch of 3
+            ((input_tensor_chunk_1, ), (some_int, )),  # batch of 3
+            ((input_tensor_chunk_2, ), (some_int, )),  # batch of 3
+            ((input_tensor_chunk_3, ), (some_int, )),  # batch of 1
+        ],
+        dummy_flags = [False, False, False, False]
+
+        """
+        # Initialize lists to hold split args and kwargs
+        split_args_list: list[list] = [[] for _ in range(self.num_chunks)]
+        split_kwargs_list: list[dict] = [{} for _ in range(self.num_chunks)]
+        dummy_chunk_flags: list[bool] = []
+
+        # Helper function to process both args and kwargs
+        def process_items(items, is_kwargs=False):
```

Comment:
```suggestion
        def process_items(items, is_kwargs: bool = False) -> None:
            """Helper to add args/kwargs into above lists."""
```

This comment should be a docstring